File: src/main/java/no/poc/domain/Bruker.java
package no.poc.domain;

public class Bruker extends AbstractEntity<Long> {
    private long brukerId;
    private String fornavn;
    private String etternavn;

    public Bruker() {
        //JPA
    }

    public Bruker(long brukerId, String fornavn, String etternavn) {
        this.brukerId = brukerId;
        this.fornavn = fornavn;
        this.etternavn = etternavn;
    }

    @Override
    public Long getId() {
        return brukerId;
    }

    public long getBrukerId() {
        return brukerId;
    }

    public String getFornavn() {
        return fornavn;
    }

    public String getEtternavn() {
        return etternavn;
    }
}

File: src/main/java/no/poc/domain/Bruker.kt
package no.poc.domain

data class Bruker(
        val id: Long,
        val fornavn: String?,
        val etternavn: String?,
        val organisasjonsnavn: String?
) {
    fun isBedrift(): Boolean {
        return fornavn.isNullOrBlank()
                .and(etternavn.isNullOrBlank())
                .and(organisasjonsnavn.isNullOrBlank().not())
    }
}